services/src/main/java/org/keycloak/url/HostnameV2ProviderFactory.java

```diff
@@ -41,16 +47,23 @@ public class HostnameV2ProviderFactory implements HostnameProviderFactory, Envir
     @Override
     public void init(Config.Scope config) {
         // Strict mode is used just for enforcing that hostname is set
-        Boolean strictMode = config.getBoolean("hostname-strict", false);
+        boolean strictMode = config.getBoolean("hostname-strict", false);
 
         String hostnameRaw = config.get("hostname");
         if (strictMode && hostnameRaw == null) {
             throw new IllegalArgumentException("hostname is not configured; either configure hostname, or set hostname-strict to false");
         } else if (hostnameRaw != null && !strictMode) {
             // We might not need this validation as it doesn't matter in this case if strict is true or false. It's just for consistency – hostname XOR !strict.
 //            throw new IllegalArgumentException("hostname is configured, hostname-strict must be set to true");
+            LOGGER.info("If hostanme is specified, hostname-strict is effectively ignored");
         }
-
+        
+        List<String> inUse = REMOVED_OPTIONS.stream().filter(s -> config.get(s) != null).toList();
```

Comment:
@shawkins I believe this won't work as the config provider here has a scope that only contains options starting with `kc.spi-hostname-v2-`. The old options [had a prefix of `kc.spi-hostname-default-`](https://github.com/keycloak/keycloak/blob/25.0.6/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnameV1PropertyMappers.java#L21).

Since we don't have access to all config options through the SmallRye Config here, we have move it somewhere to quarkus module I'm afraid.